test/e2e/cypress/e2e/host_details.cy.js

```diff
@@ -359,12 +359,14 @@ context('Host Details', () => {
   describe('Deregistration', () => {
     describe('"Clean up" button should be visible only for an unhealthy host', () => {
       it('should not display the "Clean up" button for healthy host', () => {
-        cy.task('startAgentHeartbeat', [selectedHost.id]);
         cy.contains('button', 'Clean up').should('not.exist');
       });
 
       it('should show the "Clean up" button once heartbeat is lost and debounce period has elapsed', () => {
         cy.task('stopAgentsHeartbeat');
+        cy.contains(`The host ${selectedHost.hostName} heartbeat is failing.`, {
```

Comment:
Just curiosity. Why do we need this check?

Reply:
Watching the screenshot of the failing tests, it looked like the heartbeat toast was showing up still when the "Clean Up" button failed to be clicked. So basically this adds another point to sync with the state before clicking. I know it shouldn't be needed, but it does look like it helped after many tests. The long term solution is a bit of a refactor of the e2e tests that I've been already giving some thought.